### PR TITLE
Fix OkHttp bug where streams could be started after they had been can…

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
…celled

### CODE REVIEW

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  @Test
+  public void getType() {
+    assertEquals(MethodType.UNARY, stream.getType());
+  }
+
+  @Test
+  public void request() {
+    assertEquals(MethodType.UNARY, stream.getType());
```
Comment (ejona86) [inline okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java]: This is identical to getType() test.

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    stream.sendCancel(Status.CANCELLED);
+
+    assertEquals(Status.Code.CANCELLED, statusRef.get().getCode());
+  }
+
+  @Test
+  public void sendCancel_started() {
+    stream.start(new BaseClientStreamListener());
+    stream.start(1234);
```
Comment (ejona86) [inline okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java]: Oh... I didn't realize start() was now overloaded. Fun!

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java (hunk 1 above)
Comment (carl-mastrangelo) [inline okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java]: Oops, that should not be part of this PR.

**Comment by ejona86** [comment]
@carl-mastrangelo, the request() test seems off, but otherwise LGTM

File: okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java (hunk 2 above)
Comment (carl-mastrangelo) [inline okhttp/src/test/java/io/grpc/okhttp/OkHttpClientStreamTest.java]: Yeah.... it really should have another name.  start(id)  is when concurrent streams have not been reached.

**Comment by carl-mastrangelo** [comment]
Removed the request test.  Thanks for the speedy review!
